avocado/plugins/assets.py

```diff
@@ -0,0 +1,179 @@
+                call = call + '"' + ast.literal_eval(k.value) + '", '
+        # remove last comma and add parenthesis
+        call = call[:-2] + ")\n"
+
+        return call
+    # _build_call()
+
+    def visit_Call(self, node):
+        """
+        Visit calls and build call string if call matches the pattern
+        :param node: (Sub)Tree root to start the parse
+        :type node: ast.*
+        """
+        if isinstance(node.func, ast.Attribute):
+            if self.pattern in node.func.attr:
+                call = self._build_call(node)
+                if call is not None:
+                    self.calls.append(call)
+    # visit_call()
+
+    def fetcher(self):
+        """
+        Build an intrumented avocado test with all fetch_asset calls and
+        execute it.
+        """
+        # build raw code from list of calls, if available
+        if len(self.calls) > 0:
+            rcalls = ""
+            for call in self.calls:
+                rcalls = rcalls + call
+        else:
+            LOG_UI.debug('No supported assets found to fetch from %s'
+                         % self.file_name)
+            return
+
+        # temporary test source file
+        fetch_test = tempfile.NamedTemporaryFile(suffix=".py")
+        fetch_test.write(TEST_TEMPLATE.format(fetch_assets=rcalls).encode())
+        fetch_test.flush()
+
+        # run teporary test
+        LOG_UI.debug('Fetching assets from %s' % self.file_name)
+        self.result = process.run("avocado run --json - %s" % fetch_test.name,
+                                  ignore_status=True)
+        fetch_test.close()
+
+        # check if test ran without problems
+        fetch_test_stdout = json.loads(self.result.stdout)
+        if (self.result.exit_status != 0
+                or fetch_test_stdout['pass'] != fetch_test_stdout['total']):
+            LOG_UI.error('%s Check %s for more details.' % (
+                fetch_test_stdout['tests'][0]['fail_reason'],
+                fetch_test_stdout['tests'][0]['logfile']))
+    # fetcher()
+# FetchAssetHandler()
+
+
+class Assets(CLICmd):
+    """
+    Implements the avocado 'assets' subcommand
+    """
+    name = 'assets'
+    description = 'Manage assets'
+
+    def configure(self, parser):
+        """
+        Add the subparser for the assets action.
+
+        :param parser: Main test runner parser.
+        """
+        parser = super(Assets, self).configure(parser)
+
+        subcommands = parser.add_subparsers(dest='assets_subcommand')
+        subcommands.required = True
+
+        fetch_subcommand_parser = subcommands.add_parser('fetch',
+                                help='Fetch assets from test source'\
+                                ' or config file if it\'s not already in the'\
+                                ' cache')
+        fetch_subcommand_parser.add_argument('references', type=str,
+                                             default=[], nargs='*',
```

Comment:
There is no need to specify `type=str`, nor `default=[]` when `nargs='*'` is used. Also I'd recommend to use `nargs='+'` which always requires at least one argument (therefor check on line 164 would become unnecessary).